pageserver/src/tenant/metadata.rs

```diff
@@ -350,18 +380,21 @@ pub(crate) struct MetadataUpdate {
     disk_consistent_lsn: Lsn,
     prev_record_lsn: Option<Lsn>,
     latest_gc_cutoff_lsn: Lsn,
+    last_aux_file_policy: Option<AuxFilePolicy>,
 }
 
 impl MetadataUpdate {
     pub(crate) fn new(
         disk_consistent_lsn: Lsn,
         prev_record_lsn: Option<Lsn>,
         latest_gc_cutoff_lsn: Lsn,
+        last_aux_file_policy: Option<AuxFilePolicy>,
```

Comment:
Use of `TimelineMetadata` causes a lot of changes to ripple through and I didn't fully understand everything.

I think that there's an assumption that the timeline for which we configure this value will receive flush or gc (compaction is not enough) after changing the setting so that `MetadataUpdate` mechanism is used to propagate the latest value to `IndexPart`.

For this, I think it might actually be fine as far as data consistency goes. It's good that there is code to copy the setting to a new branches. However I think there might be a situation where after restart the "main" branch flips back to not using a setting which was already propagated to a child branch, and the re-ingest using old format.

Let's try to discuss this problem next week; I should have some code ready because this is essentially the "how do we make sure we only act on committed data on index_part.json" which I need to handle for timeline ancestor detach.